### PR TITLE
Upgrade cardano-binary to GHC 8.6.3

### LINKED ISSUE #263: Upgrade cardano-binary to GHC 8.6.3
To be able to integrate with other packages that are part of the Shelley rewrite, `cardano-binary` has to be upgraded from GHC 8.4.4 to GHC 8.6.3.

### PR DESCRIPTION
**Pull request by mdimjasevic**
This upgrades `cardano-binary` to GHC 8.6.3.

The `tasty-hedhehog` package had to be declared in a `stack.yaml` file.

Closes #263.

### CODE REVIEW

File: binary/stack.yaml [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
   - test
 
 extra-deps:
   - git: https://github.com/input-output-hk/cardano-prelude
-    commit: 54f2861978a02a2b04f626f47005769bf23b853b
+    commit: 3c5b1f78bb07d361ff8a1d20bf0350d0c5b115b9
     subdirs:
       - .
       - test
+  - pretty-show-1.8.2
```
Comment (ruhatch) [inline binary/stack.yaml]: This won't be necessary once you make the change in `cardano-prelude` :)

**Review by ruhatch — CHANGES_REQUESTED** [review]

File: binary/stack.yaml (hunk 1 above)
Comment (mdimjasevic) [reply binary/stack.yaml]: @ruhatch: done!

**Comment by mdimjasevic** [comment]
bors try

**Comment by mdimjasevic** [comment]
bors retry

**Comment by avieth** [comment]
Related https://github.com/input-output-hk/cardano-sl/pull/4035

I found that `micro-recursion-schemes` still has a `template-haskell < 2.14` constraint, but `recursion-schemes` latest admits `2.14`

**Comment by mdimjasevic** [comment]
bors retry

**Comment by mdimjasevic** [comment]
bors try

**Comment by mdimjasevic** [comment]
bors retry

**Comment by mdimjasevic** [comment]
@ruhatch, do you want to take another look now? I addressed the issue with `pretty-show`.

**Review by ruhatch — APPROVED** [review]

**Pull request merged**